src/ert/_c_wrappers/job_queue/forward_model.py

```diff
@@ -1,9 +1,19 @@
+import json
+import logging
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterator, List, Optional
+
 from cwrap import BaseCClass
-from ecl.util.util import StringList
 
 from ert._c_wrappers import ResPrototype
 from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob, ExtJoblist
-from ert._c_wrappers.util.substitution_list import SubstitutionList
+from ert._clib import job_kw
+
+if TYPE_CHECKING:
+    from ert._c_wrappers.enkf.substituter import Substituter
```

Comment:
so maybe i can take this opportunity and ask you to kindly shed some light on why at this particular point you have to use such an import pattern, instead of just saying
```
from ert._c_wrappers.enkf.substituter import Substituter
```
and using `Substituter` rather than the string literal `"Substituter"` further down in type annotations.

Reply:
The reason is that this import is only used for typechecking. This lowers the chance of getting circular dependencies as the type checker does a far better job at partial instantiation.

Reply:
ahh! cool, thank you! by the time the function is called that uses a var of type Substituter, the necessary code will already have been read / imported elsewhere.
thanks for the explanation!